File: src/main/java/com/aau/wizard/model/Deck.java
package com.aau.wizard.model;

import com.aau.wizard.model.enums.CardSuit;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Deck {
    private final List<ICard> cards = new ArrayList<>();

    public Deck() {
        initializeDeck();
    }

    private void initializeDeck() {
        // Standardkarten (1–13 in jeder Farbe außer SPECIAL)
        for (CardSuit cardSuit : CardSuit.values()) {
            if (cardSuit != CardSuit.SPECIAL) {
                for (int value = 1; value <= 13; value++) {
                    cards.add(CardFactory.createCard(cardSuit, value));
                }
            }
        }

        // Spezialkarten (4 Wizard + 4 Jester)
        for (int i = 0; i < 4; i++) {
            cards.add(CardFactory.createCard(CardSuit.SPECIAL, 14));// Wizard
            cards.add(CardFactory.createCard(CardSuit.SPECIAL, 0));//Jester
        }
    }

    public void shuffle() {
        Collections.shuffle(cards);
    }

    public List<ICard> draw(int amount) {
        if (amount < 1 || amount > cards.size()) {
            throw new IllegalArgumentException("Cannot draw " + amount + " cards from deck");
        }

        List<ICard> drawn = new ArrayList<>();
        for (int i = 0; i < amount; i++) {
            drawn.add(cards.remove(0));
        }
        return drawn;
    }

    public int size() {
        return cards.size();
    }
}

File: src/main/kotlin/com/aau/wizard/core/cards/Deck.kt
package com.aau.wizard.core.cards

class Deck {
    private val cards: MutableList<Card> = mutableListOf()

    init {
        initializeDeck()
    }

    private fun initializeDeck() {
        // Standard Karten (1-13 in jeder Farbe)
        Suit.values().filter { it != Suit.SPECIAL }.forEach { suit ->
            (1..13).forEach { value ->
                cards.add(Card(suit, value))
            }
        }

        //Spezialkarten (4 Zauberer + 4 Narren)
        repeat(4) {
            cards.add(Card(Suit.SPECIAL, 14)) // Wizard
            cards.add(Card(Suit.SPECIAL, 0))  // Jester
        }
    }


    fun shuffle() {
        cards.shuffle()
    }

    fun draw(amount: Int): List<Card> {
        require(amount in 1..cards.size) { "Cannot draw $amount cards from deck" }
        return List(amount) { cards.removeAt(0) }
    }

    fun size(): Int = cards.size
}